src/cmd/services/m3query/config/config.go

```diff
@@ -293,6 +294,24 @@ func (l *GlobalLimitsConfiguration) AsLimitManagerOptions() cost.LimitManagerOpt
 // PerQueryLimitsConfiguration represents limits on resource usage within a
 // single query. Zero or negative values imply no limit.
 type PerQueryLimitsConfiguration struct {
+	// MaxFetchedSeries limits the number of time series returned for any given 
+	// individual storage node per query, before returning result to query 
+	// service.
+	MaxFetchedSeries int `yaml:"maxFetchedSeries"`
+
+	// MaxFetchedDocs limits the number of index documents matched for any given 
+	// individual storage node per query, before returning result to query 
+	// service.
+	MaxFetchedDocs int `yaml:"maxFetchedDocs"`
+
+	// RequireExhaustive results in an error if the query exceeds the series limit.
```

Comment:
nit: "...  if the query exceeds the series limit" should probably say "any limit"